compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2714,6 +2714,13 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         def show(tpe: TypeRepr): String =
           Extractors.showType(using QuotesImpl.this)(tpe)
 
+      lazy val ConstantCode: Printer[Constant] = new Printer[Constant]:
+        def show(const: Constant): String = const.show
+
+      lazy val ConstantStructure: Printer[Constant] = new Printer[Constant]:
+        def show(const: Constant): String =
+          Extractors.showConstant(using QuotesImpl.this)(const)
```

Comment:
Is there usage for `ConstantStructure`? If `ConstantCode` is the only printer, maybe change `toString` and/or `show` method is enough for practical usage.

Reply:
`ConstantStructure` is mainly used for debugging.

We do not have access to the `Context` in the to `toString` which we need to handle the `ClassOf` case. In general, we tell users that the `toString` is unreliable and that they should use `show`, this should not be an exception or it may lead to confusion.